engines/director/cast.cpp

```diff
@@ -407,25 +412,45 @@ void Cast::loadCast() {
 
 	// Font Mapping
 	if (_castArchive->hasResource(MKTAG('V', 'W', 'F', 'M'), -1)) {
-		loadFontMap(*(r = _castArchive->getFirstResource(MKTAG('V', 'W', 'F', 'M'))));
+		if (_castArchive->isMovieChunk(MKTAG('V', 'W', 'F', 'M')))
```

Comment:
A suggestion for cleaning up this pattern:

Make a function `Archive::movieHasResource(uint32 tag)` which
1. If the Director version is 4.0.0 or greater, return whether or not a we have a chunk with that tag in `_movieChunks`.
2. Otherwise, we don't have a KEY* to reference, so just return `hasResource(tag)`.

Then, change `Archive::getMovieResource` so that
1. If the Director version is 4.0.0 or greater, return the chunk from `_movieChunks`.
2. Otherwise, just return `getResource(tag)`.

Then, wherever we need to load a chunk like this, we can simply do:
```
if (_castArchive->movieHasResource(TAG)) {
    r = _castArchive->getMovieResource(TAG);
    // load r
}
```

Once we add support for multiple casts in Director 5, this pattern will let us easily update this code to something like:
```
if (_castArchive->hasResourceOwnedBy(TAG, CAST_OR_MOVIE_ID)) {
    r = _castArchive->getResourceOwnedBy(TAG, CAST_OR_MOVIE_ID);
    // load r
}
```

Reply:
ok. I will add this in PR